src/clarity-angular/button-group/button-loading.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+
+import {Component} from "@angular/core";
+import {LoadingListener} from "../loading";
+
+@Component({
+    selector: "button[clrLoading]",
+    template: `
+        <span class="spinner spinner-inline" *ngIf="loading"></span>
+        <ng-content></ng-content>
+    `
+})
```

Comment:
What you're missing here is `providers: [{provide: LoadingListener, useExisting: LoadingButton}]`.

Angular can't make every component or provider available when requesting either their own class or any interface/abstract class they implement/extend. So you just have to manually register your `LoadingButton` component as a provider for the token `LoadingListener`.